shapely/strtree.py

```diff
@@ -122,28 +129,59 @@ def __reduce__(self):
         else:
             return (STRtree, (self.geometries,))
 
-    def query_items(self, geom: BaseGeometry) -> Sequence[Any]:
-        """Query for nodes which intersect the geom's envelope to get
-        stored items.
+    def query_items(
+        self, geom: BaseGeometry, predicate=None, distance=None
+    ) -> Sequence[Any]:
+        """
+        Return the index of all geometries in the tree with extents that
+        intersect the envelope of the input geometry.
 
         Items are integers serving as identifiers for an application.
 
+        If predicate is provided, a prepared version of the input geometry
+        is tested using the predicate function against each item whose
+        extent intersects the envelope of the input geometry:
+        predicate(geometry, tree_geometry).
+
+        The 'dwithin' predicate requires GEOS >= 3.10.
+
+        If geometry is None, an empty array is returned.
+
         Parameters
         ----------
-        geom : geometry object
-            The query geometry.
+        geom : Geometry
+            The envelope of the geometry is taken automatically for
+            querying the tree.
+        predicate : {None, 'intersects', 'within', 'contains', 'overlaps', 'crosses',\
+'touches', 'covers', 'covered_by', 'contains_properly', 'dwithin'}, optional
+            The predicate to use for testing geometries from the tree
+            that are within the input geometry's envelope.
+        distance : number, optional
+            Distance around the geometry within which to query the tree for the
+            'dwithin' predicate.  Required if predicate='dwithin'.
 
         Returns
         -------
-        An array of items stored in the tree.
-
+        ndarray
+            An array of indexes (or stored items) of geometries in the tree
```

Comment:
The previous version was more correct. The case of storing indexes is a specialization that will be obvious to users.

Reply:
Since indexes are the default and stored arbitrary values are optional, the phrasing in this PR seems more accurate.